### ISSUE TITLE
nw-gyp rebuild nodegit error targeting 0.34.x

### ISSUE THREAD

**Issue by hhhoney** [nwjs/nw.js#6869]
Rebuild nodegit error

(macOS Mojave 10.14.1, node 11.1.0, nwjs 0.34.3, npm 6.4.1, nw-gyp 3.6.3, node-gyp 3.8.0)

`--target=0.24.2` is OK
`--target=0.34.x` Failed

Reproduce:

`npm install nodegit`

`cd node_modules/nodegit`

`nw-gyp rebuild --target=0.34.3`

Error message:
```
../src/filter_registry.cc:58:21: error: no matching constructor for initialization of 'String::Utf8Value'
  String::Utf8Value name(info[0]->ToString());
                    ^    ~~~~~~~~~~~~~~~~~~~
```

```
../src/filter_registry.cc:147:21: error: no matching constructor for initialization of 'String::Utf8Value'
  String::Utf8Value name(info[0]->ToString());
                    ^    ~~~~~~~~~~~~~~~~~~~
```

**Comment by rogerwang** [comment]
The reference back to this issue in the upstream one was removed after transferring. So ref it again: https://github.com/nodejs/node/issues/23122

**Comment by rogerwang** [comment]
The fix is now available with build target '0.34.4' and this build: https://dl.nwjs.io/live-build/nw34/11-15-2018/2d9d6a7ad/v0.34.4/
